### ISSUE TITLE
Prisma2 CRUD: "before" and "after" hooks

### ISSUE THREAD

**Issue by eWert-Online** [graphql-nexus/nexus-plugin-prisma#541]
When using the `crud` functionality of nexus-prisma I would like to do something before or after the mutation is executed. 
Maybe this can also be used to manipulate the result, which is returned to the client.

For Example: 
I am using the `t.crud.createOneTicket();` functionality to give the client the ability to create a new ticket. 
After the ticket is created I have to create one or multiple stages, the ticket has to go through based on some calculations.

Right now the client would have to do the calculations and send multiple mutations to be able to do this. Which could lead to a corrupted application state on a flaky network.

It would be great, if the `crud` methods would provide some kind of hooks before and after they are executed. Kind of like how https://github.com/mickhansen/graphql-sequelize#options is doing it:

```
t.crud.createOneTicket({
  before: (root, args, context) => {
    // do something in here
  },
  after: (result, args, context) => {
    // do something in here
  },
});
```

This way the crud operations would become much more flexible and operations which are dependent on each other could be executed with just one network request.

**Comment by xd-luke** [comment]
Yes please!

**Comment by mohe2015** [comment]
It would be great if they would also support some authorization hook.

**Comment by BjoernRave** [comment]
I could also really use this to log the actions being taken. + 1

**Comment by cclaflin89** [comment]
+1

**Comment by Weakky** [comment]
This feature was implemented in #674. It's slightly different than this proposal, check out the PR to see the final implementation.

We've just released nexus-prisma@0.15.0-next.1.
You can try the feature by installing nexus-prisma@next right now. Feel free to report any bug related to it in here. I'll re-open the issue if needed 🙏
